util/autogen_banner.py

```diff
@@ -0,0 +1,49 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from textwrap import wrap
+
+LICENSE_BANNER = [
+    "Copyright lowRISC contributors.",
+    "Licensed under the Apache License, Version 2.0, see LICENSE for details.",
+    "SPDX-License-Identifier: Apache-2.0",
+]
+
+AUTOGEN_BANNER = [
+    "-------------------------------------------------------------------------",
+    "-------- W A R N I N G: A U T O - G E N E R A T E D   C O D E !! --------",
+    "DO NOT EDIT THIS FILE DIRECTLY. GENERATED WITH THE FOLLOWING COMMAND:",
+    "-------------------------------------------------------------------------",
+]
+
+
+def get_autogen_banner(command: str, comment=None) -> str:
+    """Returns a commented out auto-generated code warning banner.
+
+    command is a fully formatted string representing what command was used
+    to auto-generate the source.
+    comment is the style of comment supported by the file type.
+    """
+    command_lines = wrap(command.strip(), width=len(AUTOGEN_BANNER[0]))
+    banner = AUTOGEN_BANNER[:-1] + command_lines + [AUTOGEN_BANNER[-1]]
+    return apply_comment(banner, comment)
+
+
+def get_license_banner(comment=None) -> str:
```

Comment:
Why does this script care about the license header?

Reply:
Some files are auto-generated right from within the Python script. Those will need to add license banner as well. See:
https://github.com/lowRISC/opentitan/blob/366021d6f1bacab43ac310d80f2984487857db93/util/topgen.py#L44